olive/passes/onnx/insert_beam_search.py

```diff
@@ -84,9 +98,20 @@ def chain_model(
     ):
         from onnxruntime import __version__ as OrtVersion
         from onnxruntime.transformers.convert_generation import get_shared_initializers
+        from transformers import AutoTokenizer
 
         # version check
         version_1_16 = version.parse(OrtVersion) >= version.parse("1.16.0")
+        version_1_17_1 = version.parse(OrtVersion) >= version.parse("1.17.1")
+        # NOTE: will ignore cross qk related options for now
+
+        # get tokenizer
+        model_name = options["model_name"] or model_config.get("_name_or_path")
+        if not model_name:
+            raise ValueError("model_name is not provided and cannot be inferred from model attributes.")
+        if not options["model_name"]:
+            logger.info("Inferred model name from model attributes: %s", model_name)
+        tokenizer = AutoTokenizer.from_pretrained(model_name)
```

Comment:
Nit: seems only ORT >= 1.17.1 needs load tokennizer. Add a condition check here can save unnecessary load.

Reply:
Thanks! updated to only get tokenizer when needed.